jnosql-mapping/jnosql-mapping-core/src/main/java/org/eclipse/jnosql/mapping/reflection/Reflections.java

```diff
@@ -148,10 +146,17 @@ void makeAccessible(Field field) {
      * @throws ConstructorException when the constructor has public and default
      */
     public <T> Constructor<T> getConstructor(Class<T> type) {
+
+        final Predicate<String> isIdAnnotation = Id.class.getName()::equals;
+        final Predicate<String> isColumnAnnotation = Column.class.getName()::equals;
+
         final Predicate<Constructor<?>> defaultConstructorPredicate = c -> c.getParameterCount() == 0;
         final Predicate<Constructor<?>> customConstructorPredicate = c -> {
             for (Parameter parameter : c.getParameters()) {
-                if (parameter.getAnnotation(Id.class) != null || parameter.getAnnotation(Column.class) != null) {
+                if (Arrays.stream(parameter.getAnnotations())
```

Comment:
This condition increased, how about refactoring to a method that says explicitly what is going on here, such as "hasNoSQLAnnotation"